test/jasmine/assets/negate_if.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+/**
+ * Conditionally negate an expectation:
+ *
+ *   negateIf(expect(x), xIsSomethingElse).toBe(0);
+ *
+ * is equivalent to:
+ *
+ *   if(xIsSomethingElse) expect(x).not.toBe(0);
+ *   else expect(x).toBe(0);
+ */
+module.exports = function negateIf(expectation, negate) {
```

Comment:
Non-blocking but I wonder if there's a way to rewrite this a custom jasmine matcher. So that we have something like:

```js
expect(x).negateIf(negate).toBe(0)
```

Reply:
`expect(x).negateIf(negate).toBe(0)` would definitely be more readable - I'll give it a shot.